docs/ci-cd-environment/aws-support.md

```diff
@@ -0,0 +1,203 @@
+---
+
+# AWS support
+
+!!! beta "Self-hosted agents - closed beta"
+    Self-hosted agents are in closed beta. If you would like to run Semaphore agents in your infrastructure, please [contact us and share your use case](https://semaphoreci.com/contact). Our team will get back to you as soon as possible.
+
+If you intend to run your agents in AWS, the [agent-aws-stack][agent-aws-stack] can help you deploy an autoscaling fleet of agents in your AWS account.
+
+## Features
+
+- Run self-hosted agents in Linux-based machines
+- [Dynamically increase and decrease](#scaling-based-on-job-demand) the number of agents available based on your job demand
+- Deploy [multiple stacks of agents](#multiple-stacks), one for each self-hosted agent type
+- [Access the agent EC2 instances](#agent-instance-access) through SSH or using [AWS Systems Manager Session Manager][aws session manager]
+- Use an S3 bucket to [cache the dependencies][caching] needed for your jobs
+- Control the size of your agent instances and of your agent pool
+
+## Requirements
+
+The [agent-aws-stack][agent-aws-stack] is an [AWS CDK][aws cdk] application written in JavaScript, which depends on a few things to work:
+
+- Node and NPM, for building and deploying the CDK application and managing its dependencies
+- Make, Python 3 and Packer for AMI creation and provisioning
+- [AWS credentials][aws credentials] properly configured
+
+## Usage
+
+<b>1. Download the CDK application and install dependencies</b>
+
+```
+curl -sL https://github.com/renderedtext/agent-aws-stack/archive/refs/tags/v0.1.3.tar.gz -o agent-aws-stack.tar.gz
+tar -xf agent-aws-stack.tar.gz
+cd agent-aws-stack-0.1.3
+npm i
+```
+
+You can also fork and clone the repository.
+
+<b>2. Build your AMI</b>
+
+```
+make packer.build
+```
+
+This command uses packer to create an AMI with everything the agent needs in your AWS account. The AMI is based on the Ubuntu 20.04 server.
+
+<b>3. Create the encrypted SSM parameter for the agent type registration token</b>
+
+When creating your agent type through the Semaphore UI, you get a [registration token][registration token]. Since this is a sensitive piece of information, create an encrypted AWS SSM parameter with it:
+
+```
+aws ssm put-parameter \
+  --name YOUR_SSM_PARAMETER_NAME \
+  --value "YOUR_AGENT_TYPE_REGISTRATION_TOKEN" \
+  --type SecureString \
+  --key-id YOUR_KMS_KEY_ID
+```
+
+<b>4. Set environment variables</b>
+
+```
+export SEMAPHORE_AGENT_TOKEN_PARAMETER_NAME=YOUR_SSM_PARAMETER_NAME
+export SEMAPHORE_AGENT_TOKEN_KMS_KEY=YOUR_KMS_KEY_ID
+export SEMAPHORE_AGENT_STACK_NAME=YOUR_STACK_NAME
+export SEMAPHORE_ORGANIZATION=YOUR_ORGANIZATION
+```
+
+[Other environment variables](#configuration) may be configured as well, depending on your needs.
+
+<b>5. Bootstrap the CDK application</b>
+
+The AWS CDK requires a few resources to be around for it to work properly:
+
+```
+npm run bootstrap -- aws://YOUR_AWS_ACCOUNT_ID/YOUR_AWS_REGION
+```
+
+<b>6. Deploy the stack</b>
```

Comment:
Usually, it is better to have subtitles than bolded sentences.

Two benefits:
- Subtitles will appear in the TOC, so you can quickly jump to it.
- Subtitles are linkable, and you can share the link with your team